contrib/database/sql/sql.go

```diff
@@ -171,6 +172,13 @@ func (t *tracedConnector) Driver() driver.Driver {
 	return t.connector.Driver()
 }
 
+// Close closes the dbClose channel
+// This method will be invoked when DB.Close() is called: https://cs.opensource.google/go/go/+/refs/tags/go1.23.4:src/database/sql/sql.go;l=943-947
+func (t *tracedConnector) Close() error {
+	close(t.dbClose)
```

Comment:
Could we guarantee that `tracedConnector.Close()` gets called only once? If so, this looks good to me.

However, we could also built some guardrails just in case if it gets called multiple time.

Reply:
nvm, I got my answer https://cs.opensource.google/go/go/+/refs/tags/go1.23.4:src/database/sql/sql.go;l=919-923;drc=6de5a7180cc6459235895c76c792a7f15be5625d;bpv=1

It might be a good idea to add comment above the channel close that we only expect it would called once by the `DB.Close()`